iryna_kurhuzenkava/14.ansible.ws/ans_ws/roles/redmine_app/tasks/main.yaml

```diff
@@ -0,0 +1,74 @@
+---
+- name: Redmine. Install packages
+  apt:
+    name: "{{ apt_redmine_packages }}"
+    state: latest
+    update_cache: yes
+
+- name: Redmine. Clone repository
+  git:
+    repo: "{{ app_repo }}"
+    dest: "{{ app_home }}"
+    version: '{{ app_version }}'
+    accept_hostkey: yes
+    update: no
+- name: Redmine. Change permissions
+  file:
+    path: "{{ app_home }}/public"
+    owner: www-data
+    group: www-data
+    recurse: yes
+- name: Redmine. Change permissions
+  file:
+    path: "{{ app_home }}/tmp"
+    mode: '777'
+    recurse: yes
+
+
+- name: Redmine. Setup 01
+  shell: |
+    gem install bundler
+    bundle update   
+  args:
+    executable: /bin/bash
+    chdir: "{{ app_home }}"
+  tags: 
+    - setup
+
+- name: Session store secret generation
+  shell: rake generate_secret_token
+  args:
+   chdir: "{{ app_home }}"
+   creates: "{{ app_home }}/config/initializers/secret_token.rb"
+  tags:
+    - setup
+
+- name: Redmine. Setup 02
+  shell: | 
+    bundle exec rake db:migrate RAILS_ENV=production 
+    bundle exec rake redmine:plugins:migrate RAILS_ENV=production
+    bundle install --without development test
+    RAILS_ENV=production rake db:migrate
+    RAILS_ENV=production REDMINE_LANG='{{ app_lang }}' rake redmine:load_default_data
+  args:
+    chdir: "{{ app_home }}"
+  tags:
+    - setup
+
+- name: Configuration files for virtualhost
+  template:
+    src: redmine.site.conf.j2
+    dest: "/etc/apache2/sites-enabled/redmine.site.conf"
+  notify:
+    - apache restart 
+
+- name: Config database
```

Comment:
Why is this here? Please put in correct place. You do deployment of application, and in the end of deployment copy database connection settings. This is wrong